frame/quorum/src/lib.rs

```diff
@@ -695,6 +695,14 @@ pub mod pallet {
     // Record the vote in the storage
     fn commit_vote(who: T::AccountId, proposal_id: Hash, in_favour: bool) -> DispatchResult {
       let block_number = T::Security::get_current_block_count();
+      ensure!(
+        Self::proposals().into_iter().find(|prop| {
+          let (id, _block_number, _proposal) = prop;
+          *id == proposal_id
+        }) != None,
+        Error::<T>::ProposalDoesNotExist
+      );
+
```

Comment:
For better readability I would go with something like this;

```rust
ensure!(
  Self::proposals()
    .into_iter()
    .find(|(id, _, _)| *id == proposal_id)
    .is_some(),
  Error::<T>::ProposalDoesNotExist
);
```

Reply:
Good point, updated :)